### PR TITLE
samples: subsys: video: add arducam mega camera full-featured sample

### PR DESCRIPTION
**Pull request by ArduCAM**
This example is a full-featured sample for the Arducam Mega camera driver. By using serial communication with a PC software, it can capture images, display them, and also control camera settings such as exposure and gain.

### CODE REVIEW

**Comment by ArduCAM** [comment]
@kartben 
Hello, we are sending mega API sample to zephyr, and we have no idea about how to solve the issue.
 
image

File: samples/subsys/video/mega_full-featured/sample.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+sample:
+  name: arducam mega camera full-featured test
+tests:
+  sample.video.test:
+    build_only: true
+    tags: video
+    platform_allow:
+      - rpi_pico
+    depends_on: video
```
Comment (kartben) [inline samples/subsys/video/mega_full-featured/sample.yaml:9]: rpi_pico doesn't have any built-in video device so it's not setting "video" as one of the supported features 
```suggestion
```

File: samples/subsys/video/mega_full-featured/sample.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+sample:
+  name: arducam mega camera full-featured test
+tests:
+  sample.video.test:
```
Comment (kartben) [inline samples/subsys/video/mega_full-featured/sample.yaml:4]:
```suggestion
  sample.video.arducam_mega:
```

File: samples/subsys/video/mega_full-featured/prj.conf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+CONFIG_PRINTK=y
+CONFIG_LOG=y
+CONFIG_SERIAL=y
+
+CONFIG_SPI=y
+CONFIG_VIDEO=y
+CONFIG_VIDEO_INIT_PRIORITY=72
+CONFIG_VIDEO_BUFFER_POOL_SZ_MAX=2048
+CONFIG_VIDEO_BUFFER_POOL_NUM_MAX=2
+CONFIG_VIDEO_ARDUCAM_MEGA=y
```
Comment (kartben) [inline samples/subsys/video/mega_full-featured/prj.conf:10]:
```suggestion
```

File: samples/subsys/video/mega_full-featured/README.rst [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+- `Arducam Mega camera module`_
+
+Wiring
+******
+
+On :ref:`rpi_pico`, The Arducam Mega camera module should be connected SPI0.
+The PC should be connected to UART0 using a USB cable.
+
+   .. note:: Be careful during connection!
+    Use separate wires to connect SPI pins with pins on the rpi_pico board.
```
Comment (kartben) [inline samples/subsys/video/mega_full-featured/README.rst:29]: wrong indentation for the note block -- you may check rendered doc here: https://builds.zephyrproject.io/zephyr/pr/67538/docs/samples/subsys/video/mega_full-featured/README.html

**Review by kartben — CHANGES_REQUESTED** [review]
Small comment to get you going with the CI fail, this will need further review. BTW I am not sure a sample for a specific video device should live in the subsys/video folder, but probably rather in samples/drivers/video. WDYT @loicpoulain?

**Comment by ArduCAM** [comment]
We will merge it to arducam mega driver PR

**Pull request closed without merging**